haskell-src/Concordium/Types.hs

```diff
@@ -1219,6 +1174,30 @@ instance S.Serialize TokenParameter where
         S.putWord32be (fromIntegral (BSS.length parameter))
         S.putShortByteString parameter
 
+-- | A wrapper type for (de)-serializing an CBOR-encoded initialization parameter to/from JSON.
+--  This can parse either an JSON object representation of 'TokenInitializationParameters'
+--  (which is then re-encoded as CBOR) or a hex-encoded byte string. When rendering JSON,
+--  it will render as a JSON object if the contents can be decoded to a
+-- 'TokenInitializationParameters', or otherwise as the hex-encoded byte string.
+newtype EncodedTokenInitializationParameters = EncodedTokenInitializationParameters TokenParameter
```

Comment:
To use this wrapper, we need to use it instead of the `TokenParameter` here:

```
data CreatePLT = CreatePLT
    { -- | The symbol of the token.
      _cpltTokenSymbol :: !TokenId,
      -- | A SHA256 hash that identifies the token module implementation.
      _cpltTokenModule :: !TokenModuleRef,
      -- | The address of the account that will govern the token.
      _cpltGovernanceAccount :: !AccountAddress,
      -- | The number of decimal places used in the representation of amounts of this token. This determines the smallest representable fraction of the token.
      _cpltDecimals :: !Word8,
      -- | The initialization parameters of the token, encoded in CBOR.
      _cpltInitializationParameters :: !EncodedTokenInitializationParameters
    }
```

Reply:
I have fixed this - my idea is not to expose the wrapper type directly, but just to use it in the ToJSON and FromJSON definitions for CreatePLT.